### ISSUE TITLE
Builds currently failing for openssl-3.0 branch

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#18945]
The HEAD of the openssl-3.0 branch is currently failing during build:

````
crypto/evp/evp_lib.c:536:32: error: 'EVP_CIPHER_CTX' {aka 'struct evp_cipher_ctx_st'} has no member named 'iv_len'
  536 |         ((EVP_CIPHER_CTX *)ctx)->iv_len = len;
      |                                ^~
crypto/evp/evp_lib.c:538:12: error: 'len' undeclared (first use in this function); did you mean 'mblen'?
  538 |     return len;
      |            ^~~
      |            mblen
crypto/evp/evp_lib.c:538:12: note: each undeclared identifier is reported only once for each function it appears in
crypto/evp/evp_lib.c:539:1: error: control reaches end of non-void function [-Werror=return-type]
  539 | }
      | ^
cc1: all warnings being treated as errors
````

f428e2112c6 is probably the culprit from #18875.

**Comment by tom-cosgrove-arm** [comment]
Should the CI have caught this?

**Comment by mattcaswell** [comment]
Fixed by #18875.

**Comment by mattcaswell** [comment]
> Should the CI have caught this?

The per PR CIs only run on the target branch (which was master in this case). We have separate CIs that run on push - and those did pick up this problem (which is how I knew about it).
